### PR TITLE
GIX-1212: Filter proposals by reward status

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Users want to be able to filter SNS proposals by reward status.

# Changes

* Apply the filter by reward status in `snsFilteredProposalsStore`.

# Tests

* Add test for reward status in snsFilteredProposalsStore.
* Add test for reward status in SnsProposals page.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UX:

https://user-images.githubusercontent.com/4550653/235697013-0d1d88da-665a-4dca-9c87-7c71f337fda3.mov

File: frontend/src/lib/utils/sns-proposals.utils.ts [hunk 1]
```diff
@@ -227,7 +227,7 @@ export const snsRewardStatus = ({
   wait_for_quiet_state,
   is_eligible_for_rewards,
 }: SnsProposalData): SnsProposalRewardStatus => {
-  if (reward_event_round > 0) {
+  if (reward_event_round > BigInt(0)) {
```
Comment (mstrasinskis) [inline frontend/src/lib/utils/sns-proposals.utils.ts:230]: Nth: `0n` also for tests

File: frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts [hunk 2]
```diff
@@ -131,4 +135,63 @@ describe("snsFilteredProposalsStore", () => {
...
+    ];
+    snsFiltersStore.setRewardStatus({
+      rootCanisterId,
+      rewardStatus,
+    });
+
+    expect(
+      get(snsFilteredProposalsStore)[rootCanisterId.toText()].proposals
+    ).toHaveLength(1);
+  });
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts:196]: Probably also check for `id`. Not critical.

**Review by mstrasinskis — APPROVED** [review]
LGTM, thanks! Just a couple of optional suggestions.

File: frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts:196]: Good point!

File: frontend/src/lib/utils/sns-proposals.utils.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/utils/sns-proposals.utils.ts:230]: resolvig to be able to merge, but let's continue the conversation

**Pull request merged**